be/src/exec/pipeline/adaptive/collect_stats_context.cpp

```diff
@@ -0,0 +1,295 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exec/pipeline/adaptive/collect_stats_context.h"
+
+#include <utility>
+
+#include "column/chunk.h"
+#include "common/statusor.h"
+#include "exec/pipeline/adaptive/adaptive_dop_param.h"
+#include "exec/pipeline/adaptive/utils.h"
+
+namespace starrocks::pipeline {
+
+/// BlockState.
```

Comment:
add some comment to explain to help us to understand the motivation
BlockState->PassthroughState
BlockState->RoundRobinState

Reply:
Some comments have already been added in `collect_stats_context.h`.